micrometer-test/src/main/java/io/micrometer/core/instrument/step/PollingAwareMockStepClock.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.step;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.util.TimeUtils;
+
+/**
+ * A clock meant to be used for testing {@link StepMeterRegistry}. This clock does the
+ * {@link StepMeterRegistry#pollMetersToRollover()} whenever the step is crossed thus
+ * simulating the expected behaviour of step meters.
+ */
+public class PollingAwareMockStepClock implements Clock {
```

Comment:
I'm not sure how meaningful or noticed it has been in the lifetime of Micrometer, but we do have an `@Incubating` annotation that I think we should put on this class in case it makes sense to make some breaking changes (or even remove) it at some point.

Reply:
Yeah. It completely make sense to add something like that and introduce future changes on this and mature this. I will work on this today evening my time and make sure that I add a commit by EOD today.

On the same note, I think it is now essential that all the StepRegistry tests uses this clock or the StepVerifier you proposed, so that this poll behaviour is not excluded in the tests. Since, the behaviour is a scheduled thread it is easy to jot account for this tests and miss out on something like what this issue tries to solve. I will probably open an separate issue to make a change like that.